pkg/controller/pod.go

```diff
@@ -52,47 +54,49 @@ var _ reconcile.Reconciler = &ReconcilePod{}
 
 func (r *ReconcilePod) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
 	var pod corev1.Pod
-
-	podKey := getQualifiedName(request.Namespace, request.Name)
+	klog.V(5).Info(fmt.Sprintf("Reconcile pod %s.", request.Name))
+	podKey := getUniqueNamespacedName(request.Namespace, request.Name)
 	if err := r.client.Get(ctx, request.NamespacedName, &pod); err != nil {
 		// if the pod has been deleted, remove entry from podToClaimsMap and claimToPodMap
 		if errors.IsNotFound(err) {
+			klog.V(5).Info(fmt.Sprintf("Failed to reconcile pod %s. Pod was deleted.", request.Name))
 			r.controllerSharedState.deletePod(podKey)
 			return reconcile.Result{}, nil
 		}
-		klog.Errorf("failed to get Pod (%s): %v", podKey, err)
+		klog.Errorf(fmt.Sprintf("Error getting the pod %s. Error: %v", podKey, err))
 		return reconcile.Result{Requeue: true}, err
 	}
 
-	// update the shared map
-	podLock := r.controllerSharedState.addPod(&pod, acquireLock)
-	defer podLock.Unlock()
+	r.controllerSharedState.addPod(&pod, acquireLock)
 
-	// create replica batches
-	if err := r.createReplicas(ctx, &pod); err != nil {
+	if err := r.createReplicas(ctx, podKey); err != nil {
+		klog.V(5).Info(fmt.Sprintf("Error creating replicas for pod %s. Error: %v. Requeuing reconciliation.", request.Name, err))
 		return reconcile.Result{Requeue: true}, err
 	}
+	klog.V(5).Info(fmt.Sprintf("Successfully created replicas for pod %s. Reconciliation succeeded.", request.Name))
 	return reconcile.Result{}, nil
 }
 
-func (r *ReconcilePod) createReplicas(ctx context.Context, pod *corev1.Pod) error {
-	if pod == nil {
-		return nil
-	}
-	podKey := getQualifiedName(pod.Namespace, pod.Name)
+func (r *ReconcilePod) createReplicas(ctx context.Context, podKey string) error {
+	klog.V(5).Info(fmt.Sprintf("Creating replicas for pod %s.", podKey))
 	volumes, err := r.controllerSharedState.getVolumesFromPod(podKey)
 	if err != nil {
+		klog.V(5).Info(fmt.Sprintf("Error getting volumes for pod %s. Error: %v", podKey, err))
 		return err
 	}
+	klog.V(5).Info(fmt.Sprintf("Pod %s has %d volumes. Volumes: %v", podKey, len(volumes), volumes))
 
-	nodes, err := r.controllerSharedState.getNodesForReplica(ctx, r, volumes, podKey)
+	nodes, err := getRankedNodesForReplicaAttachments(ctx, r, volumes)
 	if err != nil {
+		klog.V(5).Info(fmt.Sprintf("Error getting nodes for replicas for pod %s. Error: %v", podKey, err))
 		return err
 	}
 
+	// creating replica attachments for each volume
 	for _, volume := range volumes {
-		azVolume, err := azureutils.GetAzVolume(ctx, r.client, r.azVolumeClient, volume, r.namespace, true)
+		azVolume, err := azureutils.GetAzVolume(ctx, r.client, r.azVolumeClient, volume, consts.AzureDiskCrdNamespace, true)
 		if err != nil {
+			klog.V(5).Info(fmt.Sprintf("Error getting azvolumes for pod %s. Volumes: %s, Namespace: %s. Actual Namespace: azure-disk-csi. Err: %v", podKey, volume, r.namespace, err))
```

Comment:
Maybe add `namespace` field to `SharedState? as it might be mislead readers to think that the pod controller operate in a single namespace if we add `namespace` field to `ReconcilePod`?